plugin/storage/es/spanstore/reader.go

```diff
@@ -201,16 +201,13 @@ func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) tim
 	}
 }
 
-func getSourceFn(archive bool, maxDocCount int) sourceFn {
+func getSourceFn(maxDocCount int) sourceFn {
 	return func(query elastic.Query, nextTime uint64) *elastic.SearchSource {
-		s := elastic.NewSearchSource().
+		return elastic.NewSearchSource().
 			Query(query).
-			Size(maxDocCount)
-		if !archive {
-			s.Sort("startTime", true).
-				SearchAfter(nextTime)
-		}
-		return s
+			Size(maxDocCount).
+			Sort("startTime", true).
+			SearchAfter(nextTime)
```

Comment:
@yurishkuro I tried this change to have the archive behaviour be the same as the primary for obtaining the source function. Looks like it doesn't work because of the `SearchAfter` which relies on the results of the previous page to get the next page. Any thoughts on how to proceed here?

Reply:
I am not sure, need better understanding of what it's doing.

Reply:
is this resolved?

Reply:
Yeah - it is. It was related to findings in https://github.com/jaegertracing/jaeger/pull/6490#issuecomment-2585474067.